pkg/errs-dictionary/errdict.go

```diff
@@ -0,0 +1,120 @@
+	ActionContactSupport                       = "contact support if using managed Istio. Otherwise check if the issue is already fixed in " +
+		"another version of Istio at https://github.com/istio/istio/issues, raise an issue, or ask for help in " +
+		"discuss.istio.io."
+
+	// LikelyCause
+	LikelyCauseAPIServer     = "a problem with the k8s API server"
+	LikelyCauseConfiguration = "an incorrect or badly formatted configuration"
+	LikelyCauseSoftwareBug   = "an issue with the Istio code"
+	LikelyCauseTransience    = "If the error occurred immediately after installation, it is likely permanent."
+
+	// Specific errors, also applicable all over the code base.
+
+	// FailedToGetObjectFromAPIServer
+	ImpactFailedToGetObjectFromAPIServer = "If the error is transient, the impact is low. If permanent, updates for " +
+		"the objects cannot be processed leading to an out of sync control plane."
+	LikelyCauseFailedToGetObjectFromAPIServer = "The most likely cause is a problem with the k8s API server - high " +
+		"load or a transient state."
+)
+
+var (
+	GeneralFailedToGetObjectFromAPIServer = &structured.Error{
+		MoreInfo: "Failed to get an object from the k8s API server, because of a transient " +
+			"error or the object no longer exists.",
+		Impact:      ImpactFailedToGetObjectFromAPIServer,
+		Action:      ActionIfErrPersistsContactSupport,
+		LikelyCause: formatCauses(LikelyCauseFailedToGetObjectFromAPIServer) + " " + LikelyCauseTransience,
+	}
+)
+
+// Operator specific
+const (
+	// NoUpdates
+	OperatorImpactNoUpdates = "In this error state, changes to the IstioOperator CR will not result in the Istio " +
+		"control plane being updated."
+)
+
+var (
+	OperatorFailedToGetObjectInCallback = &structured.Error{
+		MoreInfo: "This error occurred because a k8s update for an IstioOperator CR did not " +
+			"contain an IstioOperator object.",
+		Impact:      OperatorImpactNoUpdates,
+		Action:      ActionIfErrPersistsContactSupport,
+		LikelyCause: formatCauses(LikelyCauseAPIServer) + " " + LikelyCauseTransience,
+	}
+	OperatorFailedToAddFinalizer = &structured.Error{
+		MoreInfo: "This error occurred because a finalizer could not be added to the IstioOperator CR. The " +
+			"controller uses the finalizer to temporarily prevent the CR from being deleted while the Istio control " +
+			"plane is being deleted. ",
+		Impact: "The error indicates that when the IstioOperator CR is deleted, the Istio control plane may " +
+			"not be fully removed.",
+		Action:      "Delete and re-add the IstioOperator CR. " + ActionIfErrPersistsContactSupport,
+		LikelyCause: formatCauses(LikelyCauseAPIServer),
+	}
+	OperatorFailedToRemoveFinalizer = &structured.Error{
+		MoreInfo: "This error occurred because the finalizer set by the operator controller could not be removed " +
+			"when the IstioOperator CR was deleted.",
+		Impact:      "The error indicates that the IstioOperator CR can not be removed by the operator controller.",
```

Comment:
meta comment: do we want to be so verbose/wordy? ie do we want

`impact=IstioOperator CR can not be removed by the operator controller`

or

`impact=The error indicates that the IstioOperator CR can not be removed by the operator controller.`

IMO if we want to use plain english proper sentences we should have an arbitrary blob (ie old log.Errorf but write good messages). If we are going to use structured logging lets make use of the fields instead

Reply:
good point, yes on reflection i think the wordiness doesn't help much with the actual error and just adds more text. the same can probably be said of go comment style and i've probably been brainwashed by writing too much go. let's make the guideline to be comprehensible but minimal.

Reply:
changed, ptal.